src/physics/base/ParticleState.hh

```diff
@@ -0,0 +1,39 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ParticleState.hh
+//---------------------------------------------------------------------------//
+#ifndef physics_base_ParticleState_hh
+#define physics_base_ParticleState_hh
+
+#include "ParticleDef.hh"
+#include "Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Physical (dynamic) state of a particle track.
+ *
+ * The "physical state" is just about what differentiates this particle from
+ * another (type, energy, polarization, ...) in the lab's inertial reference
+ * frame. It does not include information about the particle's direction or
+ * position, nor about path lengths or collisions.
+ *
+ * The energy is with respect to the lab frame. The particle state is
+ * immutable: collisions and other interactions should return changes to the
+ * particle state.
+ */
+struct ParticleState
```

Comment:
Isn't what Geant typically called a Track State?
Why is it immutable?
Is the current set (3 values, so I guess, quick-ish to copy/duplicate) the intended long term size or would that be extended?  For example other 'state of the track' includes position and direction, step information and caching of navigation (path) information.  The increase may flip the run-time advantage from always-copy to update-in-place.  [Also the rate of change/timing of those values are different]